### ISSUE TITLE
TypeSpec integration with JSON Schema

### ISSUE THREAD

**Issue by mikekistler** [microsoft/typespec#2464]
Services that conform to industry standard data formats, e.g Health Insights and the FHIR standard, would benefit from TypeSpec support to directly import JSON Schema definitions.

**Comment by allenjzhang** [comment]
Different requirements:
- @useRef for emitter to generate correct output which we currently support

- true import of JSON schema into TypeSpec models for in-depth application

**Comment by asaflevi-ms** [comment]
@mikekistler, @allenjzhang 
We need to be able to reference FHIR objects as external models.
As for clients, FHIR as its own serializers, so it will be great if emitters can give the option to set (by decoration, custom configuration) specific serializers.

**Comment by asaflevi-ms** [comment]
BTW - _"@useref for emitter to generate correct output which we currently support"_
I tried to use it and got "unknown decorator"

**Comment by bterlson** [comment]
I assume we are talking about Json Schema not OpenAPI here, and yeah `useRef` is not implemented there but should be. I'm not clear if this meets the "directly import" criteria in the OP though.

**Comment by bterlson** [comment]
> it will be great if emitters can give the option to set (by decoration, custom configuration) specific serializers.

Can you give more detail here? I'm not sure I understand the scenario yet.

**Comment by mikekistler** [comment]
> BTW - "@useref for emitter to generate correct output which we currently support"
> I tried to use it and got "unknown decorator"

You must import the @typespec/openapi3 package to use the `@useRef` decorator.

See [this link](https://github.com/mikekistler/typespec-openapiv3-examples/blob/main/README.md#external-ref) for more information and a Playground example of using the `@useRef` decorator to get an external reference.

**Comment by asaflevi-ms** [comment]
@mikekistler 
Is it possible to import openapi3 and generate OpenAPI version 2 json spec ?

**Comment by mikekistler** [comment]
Yes ... for OpenAPI v2 you need to use the @azure-tools/typespec-autorest package.

Here's a [playground example](https://cadlplayground.z22.web.core.windows.net/cadl-azure/?c=aW1wb3J0ICJAdHlwZXNwZWMvaHR0cCI7CskZYXp1cmUtdG9vbHMvyCUtYXV0b3Jlc3QiOwoKdXNpbmcgVHlwZVNwZWMuSHR0cDvHFUHHJzsKCkBzZXJ2aWNlKHsKICB0aXRsZTogIkV4dGVybmFsIFJlZiIsCiAgdmVyc2lvbjogIjAuNDguMSIsCn0pCm5hbWVzcGFjZSDIMVJlZjsKCm1vZGVsIFdpZGdldCDEWEB2aXNpYmlsaXR5KCJyZWFkIikKICBpZDogc3RyaW5nOwogIGRlc2NyaXB0xWfKF3dlaWdodDogZmxvYXQzMjsKfQoKQGVycm9yCkB1c2VSZWYoImNvbW1vbi5qc29uI2NvbXBvbmVudHMuc2NoZW1hcy5FxC4iKecAl8UOIHvER%2BQAn29wIOQAkSgpOugAsXzGIzs%3D&e=%40azure-tools%2Ftypespec-autorest&options=%7B%7D)

**Comment by mikekistler** [comment]
What extension/tooling is generating this error?

I think it is likely that whatever it is assumes that the $ref value is a "URL" -- that will actual resolve in an HTTP request.

I tried this same thing but using a true "URL" and I'm not getting an error:

<img width="983" alt="image" src="https://github.com/microsoft/typespec/assets/85643503/0048b177-9b77-4139-a7c2-16f23099fa02">

**Comment by allenjzhang** [comment]
While this has been an `epic` to start with, it is now a design discussion on a feature request. Removing `Epic` label.
